### PR TITLE
feat(execution): moved global max step limit to input validation

### PR DESCRIPTION
**Pull request by dorimedini-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1415)

### CODE REVIEW

**Review by dorimedini-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1415)* status: 0 of 3 files reviewed, 1 unresolved discussion (waiting on @giladchase)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-NpZ6zpzBTUEgpm_hpRo:-NpZ6zpzBTUEgpm_hpRp:b16csm) (no related file):*
[python side](https://reviewable.io/reviews/starkware-industries/starkware/33723)

___

**Review by giladchase — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1415)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @dorimedini-starkware)

___
*[`crates/blockifier/src/execution/entry_point.rs` line 227 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-NpZFxLf6jmDP-WdO0e8:-NpZFxLf6jmDP-WdO0e9:undefined) ([raw file](https://github.com/starkware-libs/blockifier/blob/c9a39d63d6a0d3192fd6859cf1ed382f95fe1dd8/crates/blockifier/src/execution/entry_point.rs#L227)):*
> ```XML
>                 .invoke_tx_max_n_steps
>                 .try_into()
>                 .expect("Failed to convert validate_max_n_steps (u32) to usize."),
> ```





_Suggestion:_
```XML
    .expect("Failed to convert invoke_tx_max_n_steps (u32) to usize."),
```
___
*[`crates/native_blockifier/src/errors.rs` line 78 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-NpZHDCy2kyG1LO3nQ35:-NpZHDCz9pM82ZppRLQ6:b-e5ztbh) ([raw file](https://github.com/starkware-libs/blockifier/blob/c9a39d63d6a0d3192fd6859cf1ed382f95fe1dd8/crates/native_blockifier/src/errors.rs#L78)):*
> ```XML
>     InvalidMaxStepsPerTx(u32),
>     #[error("Invalid max validate steps per tx: {0}.")]
>     InvalidMaxValidateStepsPerTx(u32),
> ```

To stay consistent with other errors of this kind who use "out of range", here and in SNAPI



_Suggestion:_
```XML
    #[error("Max steps per tx out of range: {0}")]
    MaxStepsPerTxOutOfRange(u32),
    #[error("Max validate steps per tx out of range: {0}")]
    MaxValidateStepsPerTxOutOfRange(u32),
```
___
*[`crates/native_blockifier/src/py_block_executor.rs` line 321 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-NpZGk2r7Twyh6-MVGP-:-NpZGk2r7Twyh6-MVGP0:b-2itjq2) ([raw file](https://github.com/starkware-libs/blockifier/blob/c9a39d63d6a0d3192fd6859cf1ed382f95fe1dd8/crates/native_blockifier/src/py_block_executor.rs#L321)):*
> ```XML
>     // Input validation.
>     if versioned_constants.invoke_tx_max_n_steps
>         > MAX_STEPS_PER_TX.try_into().expect("Failed to convert MAX_STEPS_PER_TX to u32.")
> ```

I think this is the only usage of this and the other constant, can you please make their type u64 and remove the expects?



_Code quote:_
```XML
MAX_STEPS_PER_TX.try_into().expect("Failed to convert MAX_STEPS_PER_TX to u32.")
```

**Review by dorimedini-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1415)* status: 0 of 4 files reviewed, 3 unresolved discussions (waiting on @dorimedini-starkware and @giladchase)

___
*[`crates/native_blockifier/src/errors.rs` line 78 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-NpZHDCy2kyG1LO3nQ35:-NpZu05DC-ddI4GxWIr7:b-f7atf5) ([raw file](https://github.com/starkware-libs/blockifier/blob/c9a39d63d6a0d3192fd6859cf1ed382f95fe1dd8/crates/native_blockifier/src/errors.rs#L78)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

To stay consistent with other errors of this kind who use "out of range", here and in SNAPI
</blockquote></details>

Done (with trailing period)
___
*[`crates/native_blockifier/src/py_block_executor.rs` line 321 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-NpZGk2r7Twyh6-MVGP-:-NpZtj6P2T_sR95EeUy7:b-o55s2o) ([raw file](https://github.com/starkware-libs/blockifier/blob/c9a39d63d6a0d3192fd6859cf1ed382f95fe1dd8/crates/native_blockifier/src/py_block_executor.rs#L321)):*
<details><summary><i>Previously, giladchase wrote…</i></summary><blockquote>

I think this is the only usage of this and the other constant, can you please make their type u64 and remove the expects?
</blockquote></details>

changed to u32

**Review by dorimedini-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1415)* status: 0 of 4 files reviewed, 2 unresolved discussions (waiting on @dorimedini-starkware and @giladchase)

*[a discussion](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-NpZ6zpzBTUEgpm_hpRo:-Np_pN6W7637dIypatAG:b-u4qn5q) (no related file):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

[python side](https://reviewable.io/reviews/starkware-industries/starkware/33723)
</blockquote></details>

passes CI

___

**Review by giladchase — COMMENTED** [review]
For future reference, why are we removing the global limit for users? (but not for us)
I remember it having to do with a question on slack?

> Reviewed 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-:-Nprgz1ICgcUk7pYAlxl:b-ljjxmv)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @dorimedini-starkware)

___
*[`crates/native_blockifier/src/errors.rs` line 78 at r2](https://reviewable.io/reviews/starkware-libs/blockifier/1415#-NpZHDCy2kyG1LO3nQ35:-NpogFs6840CQBCppyvQ:b6y7jr1) ([raw file](https://github.com/starkware-libs/blockifier/blob/c9a39d63d6a0d3192fd6859cf1ed382f95fe1dd8/crates/native_blockifier/src/errors.rs#L78)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

Done (with trailing period)
</blockquote></details>

Thanks!

Regarding the trailing period, our convention has been to try to not put them in errors.

This is based on various Rust guidelines, such as the [Rust development guide](https://rustc-dev-guide.rust-lang.org/diagnostics.html#diagnostic-output-style-guide), Serde [documentation](https://docs.rs/serde/1.0.106/serde/de/trait.Error.html#tymethod.custom), and Rust's [API guidelines](https://rust-lang.github.io/api-guidelines/interoperability.html?highlight=error#error-types-are-meaningful-and-well-behaved-c-good-err). The rationale, IIRC, is that error messages are frequently interpolated, leading to formatting issues like double periods (`..`) if the original error message ends with a period and is later wrapped in another error message with a trailing period.
In other words, this approach avoids redundant punctuation when errors are concatenated.

**Review by giladchase — APPROVED** [review]
> Reviewed 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/blockifier/1415)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @dorimedini-starkware)
